src/AbstractPolytope.jl

```diff
@@ -68,21 +68,9 @@ The polytope in V-representation obtained by applying the linear map ``M`` to
 the set ``P``. If the given polytope is two-dimensional, a polygon instead
 of a general polytope is returned. 
 """
-function linear_map(M::AbstractMatrix, P::AbstractPolytope{N}) where {N<:Real}
-    @assert dim(P) == size(M, 2) "a linear map of size $(size(M)) cannot be " *
-                                 "applied to a set of dimension $(dim(P))"
-
-    if dim(P) == 2
-        T = VPolygon
-    else
-        T = VPolytope
-    end
-    vlist = vertices_list(P)
-    new_vlist = Vector{Vector{N}}(undef, length(vlist))
-    @inbounds for (i, vi) in enumerate(vlist)
-        new_vlist[i] =  M * vi
-    end
-    return T(new_vlist)
+function linear_map(M::AbstractMatrix, P::AbstractPolytope{N})::VPolytope{N} where {N<:Real}
+    @assert dim(P) == size(M, 2)
+    return broadcast(v -> M * v, vertices_list(P)) |> VPolytope{N}
```

Comment:
Is there a difference between
`broadcast(v -> M * v, vertices_list(P)) |> VPolytope{N}`
and
`VPolytope{N}(broadcast(v -> M * v, vertices_list(P)))` (used for polygons)?
(I do not know `|>`.)
I would stick to one of them for consistency.

Reply:
Does the `VPolytope` output type not contradict with the name of this PR? :stuck_out_tongue:

Reply:
🔙  How about not changing the given set type by default, but giving the option of return type as a keyword argument?

Reply:
> (I do not know |>.)

it is the "pipe" operator that can be used to chaining function calls. i used it here because it seemed more clear in the code to me.

```julia
help?> |>
search: |>

  |>(x, f)

  Applies a function to the preceding argument. This allows for easy function
  chaining.

  julia> [1:5;] |> x->x.^2 |> sum |> inv
  0.01818181818181818
```

Reply:
> How about not changing the given set type by default, but giving the option of return type as a keyword argument?

We do not have constructors from vertices for the other types. And some `AbstractPolytope` subtypes are not even closed under linear map (e.g., `BallInf`), so in general we have to change the type.
So I think that we should stick to `VPolytope`. We can still add more specific implementations as usual, like we have for `Zonotope`s.

Reply:
> And some AbstractPolytope subtypes are not even closed under linear map (e.g., BallInf), so in general we have to change the type.

good point. i changed the title of this PR 😄 

how about `linear_map` of `AbstractPolygon`? in that case we can add the optional conversion argument, because the subtypes are closed under linear map.

Reply:
Okay for me, more work for you :wink: